packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  ExperimentReportResultDimension,
+  ExperimentReportVariation,
+} from "back-end/types/report";
+import React, { useMemo } from "react";
+import { FaFileExport } from "react-icons/fa";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { ExperimentTableRow, getRisk } from "../../services/experiments";
+import { Parser } from "json2csv";
+import { MetricInterface, MetricStats } from "back-end/types/metric";
+
+type CsvRow = {
+  date?: string;
+  dimension?: string;
+  metricName?: string;
+  variantName?: string;
+  riskOfChoosing?: number;
+  users?: number;
+  count?: number;
+  conversionRate?: number;
+  chanceToBeatControl?: number | null;
+  percentChange?: number | null;
+};
+
+type Variation = {
+  buckets?: { x: number; y: number }[];
+  ci?: [number, number];
+  cr: number;
+  risk?: [number, number];
+  stats?: MetricStats;
+  users: number;
+  value: number;
+};
+
+type Row = {
+  label?: string;
+  metric?: MetricInterface;
+  rowClass?: string;
+  variations?: Variation[];
+};
+
+export default function ResultsDownloadButton({
+  results,
+  metrics,
+  variations,
+  trackingKey,
+  dimension,
+}: {
+  results?: ExperimentReportResultDimension[];
+  metrics: string[];
+  variations: ExperimentReportVariation[];
+  trackingKey: string;
+  dimension: string;
+}) {
+  const { getMetricById, getDimensionById, ready } = useDefinitions();
+
+  const json2csvParser = new Parser();
+
+  const csvRows: CsvRow[] = [];
+
+  const dimensionName = getDimensionById(dimension);
+
+  if (dimension === "pre:date") {
+    results.sort((a, b) => (a.name > b.name ? 1 : b.name > a.name ? -1 : 0));
+  }
+
+  const rows = useMemo<ExperimentTableRow[]>(() => {
+    const rowsArr = [];
+    if (!results || !variations || !ready) return [];
+    results.forEach((result) => {
+      metrics.forEach((m) => {
+        const row: Row = {};
+        const metric = getMetricById(m);
+        (row.label = metric?.name),
+          (row.metric = metric),
+          (row.rowClass = metric?.inverse ? "inverse" : ""),
+          (row.variations = result.variations.map((v) => {
+            return v.metrics[m];
+          })),
+          rowsArr.push(row);
```

Comment:
Using the comma operator like this is a little unusual. It works, but I would probably just do separate statements separated by semicolons.

Reply:
I wish I had a good explanation for why I did it this way, but... I don't lol. I've refactored to use semicolons.